aiohttp/base_protocol.py

```diff
@@ -0,0 +1,64 @@
+import asyncio
+
+from .log import internal_logger
+
+
+class BaseProtocol(asyncio.Protocol):
+    def __init__(self, loop=None):
+        if loop is None:
+            self._loop = asyncio.get_event_loop()
+        else:
+            self._loop = loop
+        self._paused = False
+        self._drain_waiter = None
+        self._connection_lost = False
+        self.transport = None
```

Comment:
Do we want transport as public attribute or instead public read-only property? I think transport is only valid to set via `connection_made` method, not somehow adhoc'y.

Reply:
I'm trying don't touch transport attr, many aiohttp codes use it internally.
Tests modify `.transport` wildly as well.
Better to deprecate `.protocol` at all in our public API.

Reply:
:+1: